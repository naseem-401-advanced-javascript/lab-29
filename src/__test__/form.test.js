/* eslint-disable no-unused-vars */
import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Form from '../components/form.js';

Enzyme.configure({adapter: new Adapter() });

describe('<Form />', ()=>{
  it('can render Form',()=>{
    let app = mount(<Form/>);
    let form = app.find('form');
    expect(form.exists()).toBeTruthy();
  });

  it('can render to the DOM',()=>{
    const rendered = renderer.create(<Form />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('can GET', () => {
    let app = mount(<Form />);
    let button = app.find('button');
    button.simulate('click');
    expect(app.state('body')).toEqual({});
  });
});