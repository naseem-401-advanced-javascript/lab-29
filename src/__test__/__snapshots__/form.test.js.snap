// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> can render to the DOM 1`] = `
<main>
  <section>
    <form
      onSubmit={[Function]}
    >
      <section
        id="border"
      >
        <input
          className="url"
          name="url"
          onChange={[Function]}
          placeholder="URL"
          type="text"
          value=""
        />
        <div
          className="methods"
        >
          <label
            className="nn"
          >
            <input
              name="method"
              onClick={[Function]}
              type="radio"
              value="get"
            />
            <span>
              GET
            </span>
          </label>
          <label
            className="nn"
          >
            <input
              name="method"
              onClick={[Function]}
              type="radio"
              value="post"
            />
            <span>
              POST
            </span>
          </label>
          <label
            className="nn"
          >
            <input
              name="method"
              onClick={[Function]}
              type="radio"
              value="put"
            />
            <span>
              PUT
            </span>
          </label>
          <label
            className="nn"
          >
            <input
              name="method"
              onClick={[Function]}
              type="radio"
              value="delete"
            />
            <span>
              DELETE
            </span>
          </label>
          <label>
             
            <button
              onClick={[Function]}
              type="submit"
            >
              GO!
            </button>
             
          </label>
        </div>
      </section>
    </form>
  </section>
</main>
`;
